app/models/post.rb

```diff
@@ -248,7 +248,7 @@ def recover!
 
   # The key we use in redis to ensure unique posts
   def unique_post_key
-    "unique-post-#{user_id}:#{raw_hash}"
+    "unique-post-#{topic&.private_message? ? "pm" : "topic"}-#{user_id}:#{raw_hash}"
```

Comment:
```suggestion
    "unique#{topic&.private_message? ? "-pm" : ""}-post-#{user_id}:#{raw_hash}"
```
This will change the keys to 

`unique-pm-post-#{user_id}:#{raw_hash}` and `unique-post-#{user_id}:#{raw_hash}`.

To me this reads better and also it doesn't result in existing keys becoming invalid.